### PR TITLE
Change type of sum, min, max function column in mv

### PR DESCRIPTION
**Pull request by EmmyMiao87**
## Proposed changes
If the agg function is sum, the type of mv column will be bigint.
The only exception is that if the base column is largeint, the type of mv column will be largeint.

If the agg function is min or max, the type of mv column will be same as the type of base column.
For example, the type of mv column is smallint when the agg function is min.

Change-Id: I1b541fd03342ceba093c4aabd8a29793c52781dc

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have create an issue on [Doris's issues](https://github.com/apache/incubator-doris/issues), and have described the bug/feature there in detail
- [ ] Commit messages in my PR start with the related issues ID, like "#4071 Add pull request template to doris project"
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If this change need a document change, I have updated the document
- [ ] Any dependent changes have been merged

### CODE REVIEW

**Comment by kangkaisen** [comment]
Please add a UT to cover this PR.

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateMaterializedViewStmt.java [hunk 1]
```diff
@@ -30,11 +34,6 @@
 import org.apache.doris.common.FeNameFormat;
 import org.apache.doris.common.UserException;
```
Comment (wutiangan) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateMaterializedViewStmt.java:36]: plz import the jar package according following sequence：
4=javax
3=java
2=org
1=com
0=org.apache.doris

**Review by wutiangan — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateMaterializedViewStmt.java (hunk 1 above)
Comment (kangkaisen) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/CreateMaterializedViewStmt.java:36]: Hi, you could update the `checkstyle.xml`.

**Comment by EmmyMiao87** [comment]
> Please add a UT to cover this PR.

Added

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateMaterializedViewStmt.java (hunk 1 above)
Comment (EmmyMiao87) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/CreateMaterializedViewStmt.java:42]: Changed

**Comment by kangkaisen** [comment]
> If the agg function is sum, the type of mv column will be bigint.

@EmmyMiao87 Hi. for sum(double), sum(decimal), the type of mv column will also be bigint?

**Comment by EmmyMiao87** [comment]
> > If the agg function is sum, the type of mv column will be bigint.
> 
> @EmmyMiao87 Hi. for sum(double), sum(decimal), the type of mv column will also be bigint?

Fixed

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateMaterializedViewStmt.java [hunk 2]
```diff
@@ -353,11 +353,21 @@ public MVColumnItem buildMVColumnItem(FunctionCallExpr functionCallExpr) throws
...
         switch (functionName.toLowerCase()) {
             case "sum":
+                mvColumnName = baseColumnName;
+                mvAggregateType = AggregateType.valueOf(functionName.toUpperCase());
+                PrimitiveType baseColumnType = baseColumnRef.getType().getPrimitiveType();
+                if (baseColumnType == PrimitiveType.TINYINT || baseColumnType == PrimitiveType.SMALLINT
+                        || baseColumnType == PrimitiveType.INT) {
+                    type = Type.BIGINT;
+                } else {
+                    type = Type.fromPrimitiveType(baseColumnRef.getType().getPrimitiveType());
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateMaterializedViewStmt.java:363]: Sum float result type should be double.

**Review by kangkaisen — DISMISSED** [review]
+1

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**